### ISSUE TITLE
Body_25 caffemodel

### ISSUE THREAD

**Issue by llevkova** [CMU-Perceptual-Computing-Lab/openpose_train#3]
The provided body_25 model is not compatible with the current openpose_caffe_train. It cannot be loaded to be used as a starting point for finetuning. It gives the error: Can't parse message of type "caffe.NetParameter" because it is missing required fields: layer[0].clip_param.min, layer[0].clip_param.max. **Is it possible to release a pretrained body_25 model compatible with the d-setLayer.py generated training scripts?**

### Posting rules
1. **Duplicated posts will not be answered**. Check the [FAQ](https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/master/doc/faq.md) section, other GitHub issues, and general documentation before posting. E.g., **low-speed, out-of-memory, output format, 0-people detected, installation issues, ...**).
2. **Fill** the **Your System Configuration section (all of it or it will not be answered!)** if you are facing an error or unexpected behavior. Feature requests or some other type of posts might not require it.
3. **No questions about 3rd party libraries**:
    - Caffe errors/issues, check [Caffe](http://caffe.berkeleyvision.org) documentation.
    - CUDA check failed errors: They are usually fixed by re-installing CUDA, then re-installing the proper cuDNN version, and then re-compiling (or re-installing) OpenPose. Otherwise, check for help in CUDA forums.
    - OpenCV errors: Install the default/pre-compiled OpenCV or check for online help.
4. Set a **proper issue title**: add the Ubuntu/Windows keyword and be specific (e.g., do not call it: `Error`).
5. Only English comments.
Posts which do not follow these rules will be **ignored, closed, or reported** with no further clarification.

### Executed Command (if any)
Note: add `--logging_level 0 --disable_multi_thread` to get higher debug information.

### Type of Issue
You might select multiple topics, delete the rest:
- Compilation/installation error
- Execution error
- Help wanted
- Question
- Enhancement / offering possible extensions / pull request / etc
- Other (type your own type)

### Your System Configuration
1. **Whole console output** (if errors appeared), paste the error to [PasteBin](https://pastebin.com/) and then paste the link here: LINK

2. **OpenPose version**: Latest GitHub code? Or specific commit (e.g., d52878f)? Or specific version from `Release` section (e.g., 1.2.0)?

3. **General configuration**:
    - **Installation mode**: CMake, sh script, manual Makefile installation, ... (Ubuntu); CMake, ... (Windows); ...?
    - **Operating system** (`lsb_release -a` in Ubuntu):
    - **Operating system version** (e.g., Ubuntu 16, Windows 10, ...):
    - **Release or Debug mode**? (by default: release):
    - Compiler (`gcc --version` in Ubuntu or VS version in Windows): 5.4.0, ... (Ubuntu); VS2015 Enterprise Update 3, VS2017 community, ... (Windows); ...?

4. **Non-default settings**:
    - **3-D Reconstruction module added**? (by default: no):
    - Any other custom CMake configuration with respect to the default version? (by default: no):

5. **3rd-party software**:
    - **Caffe version**: Default from OpenPose, custom version, ...?
    - **CMake version** (`cmake --version` in Ubuntu):
    - **OpenCV version**: pre-compiled `apt-get install libopencv-dev` (only Ubuntu); OpenPose default (only Windows); compiled from source? If so, 2.4.9, 2.4.12, 3.1, 3.2?; ...?

6. If **GPU mode** issue:
    - **CUDA version** (`cat /usr/local/cuda/version.txt` in most cases):
    - **cuDNN version**:
    - **GPU model** (`nvidia-smi` in Ubuntu):

7. If **CPU-only mode** issue:
    - **CPU brand & model**:
    - Total **RAM memory** available:

8. If **Python** API:
    - **Python version**: 2.7, 3.7, ...?
    - **Numpy version** (`python -c "import numpy; print numpy.version.version"` in Ubuntu):

9. If **Windows** system:
    - Portable demo or compiled library?

10. If **speed performance** issue:
    - Report OpenPose timing speed based on [this link](https://github.com/CMU-Perceptual-Computing-Lab/openpose/blob/master/doc/speed_up_openpose.md#profiling-speed).

**Comment by gineshidalgo99** [comment]
Our models are different (and more accurate) than that one, that is why we do not pre-train on it.

**Comment by llevkova** [comment]
Thank you for your answer. Would it be possible to provide a pre-trained model body_25b which is compatible with the generated by default prototxts by your scripts?

**Comment by gineshidalgo99** [comment]
https://github.com/CMU-Perceptual-Computing-Lab/openpose_train/tree/master/experimental_models

**Comment by gineshidalgo99** [comment]
Sorry! I've just realized that the d_setLayers was not included in the repo due to some issue with our gitignore file. They are now updated!

In addition, we have added the scripts for the official body_25 from OpenPose (but we recommend using body_25b for better accuracy while maintaining the same speed).

See the updated README from https://github.com/CMU-Perceptual-Computing-Lab/openpose_train/tree/master/experimental_models

**Comment by llevkova** [comment]
This is great! Thank you so much!

**Comment by llevkova** [comment]
I have the following question and will really appreciate if you have any insights.
I trained from scratch body_25b fast model and its mAP is 51.6% on coco val2017 (17 keypoints) after running the full schedule as generated in pose_solver.prototxt.
The model you just provided has mAP of 53.2% on the same data set. Do you have an idea why I am getting a worse result? I noticed some differences in the OPData layer in the provided train proto in  experimental_models/1_25BBkg/training_results,   such as different name for the rotation angle which would not work with the current openpose code. It is possible the protos are generated with an older version of openpose? Also the variable scale_maxs: "1.5" , instead of  scale_maxs: "1.5;1.5;2.5". Is this what you recommend?

**Comment by anerisheth19** [comment]
> I have the following question and will really appreciate if you have any insights.
> I trained from scratch body_25b fast model and its mAP is 51.6% on coco val2017 (17 keypoints) after running the full schedule as generated in pose_solver.prototxt.
> The model you just provided has mAP of 53.2% on the same data set. Do you have an idea why I am getting a worse result? I noticed some differences in the OPData layer in the provided train proto in experimental_models/1_25BBkg/training_results, such as different name for the rotation angle which would not work with the current openpose code. It is possible the protos are generated with an older version of openpose? Also the variable scale_maxs: "1.5" , instead of scale_maxs: "1.5;1.5;2.5". Is this what you recommend?

What is your input resolution?

**Comment by llevkova** [comment]
The base one, 368x368

**Comment by gineshidalgo99** [comment]
Our best resuls are obtained with 4-GPU machines and a batch size of 10, training for about 800k iterations (and picking the model with maximum accuracy amount those). [This line has also been pushed into train/README.md, thanks for the question!]

**Comment by thetruejacob** [comment]
@gineshidalgo99 are the links to these experimental models down? I'm unable to download the caffe models.
